roofit/roofitcore/src/RooAbsCollection.cxx

```diff
@@ -710,30 +710,41 @@ RooAbsCollection* RooAbsCollection::selectByAttrib(const char* name, Bool_t valu
 }
 
 
-
-
 ////////////////////////////////////////////////////////////////////////////////
 /// Create a subset of the current collection, consisting only of those
 /// elements that are contained as well in the given reference collection.
-/// The caller is responsible for deleting the returned collection
+/// Returns `true` only if something went wrong.
+/// The complement of this function is getParameters().
+/// \param[in] refColl The collection to check for common elements.
+/// \param[out] outColl Output collection.
 
-RooAbsCollection* RooAbsCollection::selectCommon(const RooAbsCollection& refColl) const
+bool RooAbsCollection::selectCommon(const RooAbsCollection& refColl, RooAbsCollection& outColl) const
 {
-  // Create output set
-  TString selName(GetName()) ;
-  selName.Append("_selection") ;
-  RooAbsCollection *sel = (RooAbsCollection*) create(selName.Data()) ;
+  outColl.clear();
+  outColl.setName((std::string(GetName()) + "_selection").c_str());
 
   // Scan set contents for matching attribute
   for (auto arg : _list) {
     if (refColl.find(*arg))
-      sel->add(*arg) ;
+      outColl.add(*arg) ;
   }
 
-  return sel ;
+  return false;
 }
 
 
+////////////////////////////////////////////////////////////////////////////////
+/// Create a subset of the current collection, consisting only of those
+/// elements that are contained as well in the given reference collection.
+/// The caller is responsible for deleting the returned collection
```

Comment:
- Maybe adapt to the common style using `\param` and `\return`.

Reply:
Okay, I put this in the pipeline for the next doxygen PR!